gsplat/cuda/csrc/helpers.cuh

```diff
@@ -104,36 +110,26 @@ inline __device__ float4 transform_4x4(const float *mat, const float3 p) {
 }
 
 inline __device__ float2 project_pix(
-    const float *mat, const float3 p, const dim3 img_size, const float2 pp
+    const float2 fxfy, const float3 p_view, const float2 pp
 ) {
-    // ROW MAJOR mat
-    float4 p_hom = transform_4x4(mat, p);
-    float rw = 1.f / (p_hom.w + 1e-6f);
-    float3 p_proj = {p_hom.x * rw, p_hom.y * rw, p_hom.z * rw};
-    return {
-        ndc2pix(p_proj.x, img_size.x, pp.x), ndc2pix(p_proj.y, img_size.y, pp.y)
-    };
+    float rw = 1.f / (p_view.z + 1e-6f);
+    float2 p_proj = { p_view.x * rw, p_view.y * rw };
+    // TODO: check if 0.5 should be added to pix coords or not
```

Comment:
I think adding 0.5 to (x,y) image coordinates here is not required. However, if one later wants to convert xy to integer (u,v)'s then a 0.5 pixel length should be accounted for, but the rastarizer works with the floating point xy's so I think this should be okay.

Reply:
Good to know, thanks. Note that the previous version (the removed `ndc2pix`) had +0.5f, but in any case, the effect / error is small. Especially if camera positions and/or intrinsics are optimized, since the term can be exactly absorbed into cx, cy and approximately compensated with camera position / orientation movements.

Reply:
I also quickly tested the code both with and without +0.5f and both there were no major qualitative differences in the 3DGS results. A larger benchmark could be used to double-check this, though (e.g., are there differences in PSNR)